tests/main/interfaces-personal-files/task.yaml

```diff
@@ -0,0 +1,96 @@
+    . "$TESTSLIB/files.sh"
+
+    # Fist layer of dirs and files
+    ensure_file_exists_backup_real /root/.testfile1
+    ensure_file_exists_backup_real /root/testfile1
+    ensure_dir_exists_backup_real /root/.testdir1
+    ensure_dir_exists_backup_real /root/testdir1
+
+    # Second layer of dirs and files
+    ensure_file_exists_backup_real /root/.testdir1/.testfilé2
+    ensure_file_exists_backup_real "/root/.testdir1/test file2"
+    ensure_dir_exists_backup_real /root/.testdir1/.testdir2
+
+    # Not accessible dirs and files
+    ensure_dir_exists_backup_real /tmp/.testdir1
+    ensure_file_exists_backup_real /tmp/.testfile1
+    ensure_dir_exists_backup_real /tmp/root/
+    ensure_file_exists_backup_real /tmp/root/testfile2
+    ensure_file_exists_backup_real /tmp/root/.testfile2
+
+restore: |
+    rm -f call.error
+
+    # shellcheck source=tests/lib/files.sh
+    . "$TESTSLIB/files.sh"
+
+    clean_file /root/.testfile1
+    clean_file /root/testfile1
+    clean_dir /root/.testdir1
+    clean_dir /root/testdir1
+    clean_dir /tmp/root   
+
+execute: |
+    echo "The interface is not connected by default"
+    snap interfaces -i personal-files | MATCH "\\- +test-snapd-sh:personal-files"
+
+    echo "When the interface is connected"
+    snap connect test-snapd-sh:personal-files
+
+    echo "And not other interfaces are connected"
+    snap disconnect test-snapd-sh:home
+
+    echo "Then the snap is able to access all the files and dirs in $HOME"
+    test-snapd-sh.with-personal-files-plug -c "cat /root/.testfile1"
+    test-snapd-sh.with-personal-files-plug -c "cat /root/testfile1"
+    test-snapd-sh.with-personal-files-plug -c "ls /root/.testdir1"
+    test-snapd-sh.with-personal-files-plug -c "ls /root/testdir1"
+    test-snapd-sh.with-personal-files-plug -c "cat /root/.testdir1/.testfilé2"
+    test-snapd-sh.with-personal-files-plug -c "cat '/root/.testdir1/test file2'"
+    test-snapd-sh.with-personal-files-plug -c "ls  /root/.testdir1/.testdir2/"
+
+    echo "Then the snap is able to write just /root/.testdir1 and /root/.testfile1"
+    test-snapd-sh.with-personal-files-plug -c "echo test >> /root/.testfile1"
+    test-snapd-sh.with-personal-files-plug -c "touch /root/.testdir1/testfilé2"
+    if test-snapd-sh.with-personal-files-plug -c "echo test >> /root/testfile1" 2> call.error; then
+        echo "Expected permission error writing the personal file"
+        exit 1
+    fi
+    MATCH "Permission denied" < call.error
+
+    echo "Then the snap is not able to to access files and dirs outside $HOME"
+    test-snapd-sh.with-personal-files-plug -c "ls /tmp/.testdir1" && exit 1
+    test-snapd-sh.with-personal-files-plug -c "cat /tmp/.testfile1" && exit 1
+    test-snapd-sh.with-personal-files-plug -c "ls /tmp/root/" && exit 1
+    test-snapd-sh.with-personal-files-plug -c "cat /tmp/root/testfile2" && exit 1
+    test-snapd-sh.with-personal-files-plug -c "cat /tmp/root/.testfile2" && exit 1
```

Comment:
These should be accessible. I suspect what is happening is that you are getting ENOENT errors because while you call ensure_file_exists_backup_real, that is within the context of the global mount namespace, and not the snap namespace. If you use /var/tmp instead, it would address this (or really, `/root/.not-allowed/blah`). There is a missing `MATCH "Permission denied" < call.error` to prove that it was a permission denied and not some other error.

Reply:
I updated the test to use /var/tmp